File: Atividade_4/Animal.py

class Animal:
    def __init__(self, nome):
        self.nome = nome

    def som(self):
        raise NotImplementedError("Subclasses devem implementar este método")



class Cachorro(Animal):
    def som(self):
        return f"{self.nome} faz: Au au!"



class Gato(Animal):
    def som(self):
        return f"{self.nome} faz: Miau!"



cachorro = Cachorro("Rex")
gato = Gato("Mimi")

print(cachorro.som())  
print(gato.som())      

File: Atividade_4/Animal.go
package main

import "fmt"


type Animal interface {
    Som() string
}


type Cachorro struct {
    Nome string
}


type Gato struct {
    Nome string
}


func (c Cachorro) Som() string {
    return fmt.Sprintf("%s faz: Au au!", c.Nome)
}


func (g Gato) Som() string {
    return fmt.Sprintf("%s faz: Miau!", g.Nome)
}

func main() {
    cachorro := Cachorro{Nome: "Rex"}
    gato := Gato{Nome: "Mimi"}

    
    fmt.Println(cachorro.Som())  
    fmt.Println(gato.Som())      
}
